src/pages/studyView/charts/ChartContainer.tsx

```diff
@@ -427,15 +427,28 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
                         Compare
                     </div>
                 ),
-                onClick: () => {
-                    this.openComparisonPage(
-                        getComparisonParamsForTable(
-                            this.selectedRowsKeys,
-                            this.chartType
-                        )
-                    );
-                },
-                isDisabled: () => this.selectedRowsKeys!.length < 2,
+                onClick:
```

Comment:
this should really be just single callback and in the callback we can check to see how many selected rows there are and act accordingly.   in general we want to keep the callbacks we embed in our JSX as minimal as possible and put business logic in methods or functions.  i.e. make onClick: ()=>this.handleComparisonClick
and then in this method, we check selectedRows and forward user